baselines/common/vec_env/subproc_vec_env.py

```diff
@@ -4,51 +4,62 @@
 from .vec_env import VecEnv, CloudpickleWrapper, clear_mpi_env_vars
 
 
-def worker(remote, parent_remote, env_fn_wrapper):
+def worker(remote, parent_remote, env_fn_wrappers):
     parent_remote.close()
-    env = env_fn_wrapper.x()
+    envs = [env_fn_wrapper() for env_fn_wrapper in env_fn_wrappers.x]
+    nenvs = len(envs)
     try:
         while True:
+            buf = [None for _ in range(nenvs)]
             cmd, data = remote.recv()
             if cmd == 'step':
-                ob, reward, done, info = env.step(data)
-                if done:
-                    ob = env.reset()
-                remote.send((ob, reward, done, info))
+                for i, env in enumerate(envs):
+                    buf[i] = list(env.step(data[i]))
```

Comment:
this looks a little ugly (and , and I doubt that it gives any noticeable performance boost. 
How about:
```python
for i, env in enumerate(envs):
  ob, reward, done, info = env.step(data[i])
  if done:
    ob = env.reset()
  buf[i] = (ob, reward, done, info)
remote.send(buf)
```
or even 
```python
def step_env(env, data):
   ob, reward, done, info = env.step(data)
   if done:
       ob = env.reset()
   return (ob, reward, done, info)
remote.send([step_fn(env, data_single) for env, data_single in zip(envs, data)])
```